test/linalg/test_blas_lapack.fypp

```diff
@@ -117,6 +119,75 @@ contains
 
     end subroutine test_idamax
 
+    !> Test availability of the external BLAS interface
+    subroutine external_blas_test(error)
+        !> Error handling
+        type(error_type), allocatable, intent(out) :: error
+
+#ifdef STDLIB_EXTERNAL_BLAS           
```

Comment:
Correct me if I got this wrong, but I think the test should be calling
```
use stdlib_linalg_blas, only: axpy
...
call axpy( ... )

use stdlib_linalg_lapack, only: getrf
...
call getrf( ... )
```
Which already interface the respective routines if `STDLIB_EXTERNAL_BLAS` \ `STDLIB_EXTERNAL_LAPACK ` are set or not.

In this case, the test_blas_lapack.fypp file does not need to be moved to the C preprocessed category. Which is why the fpm build is failing as it was not included in the fypp_deployment.py file, but I'm guessing that it is not needed.

Reply:
Great comment Jose, thank you for looking into it. I've tried that. However, calling the interface does not guarantee that we're using the external library function. One thing that I had tried to check that was to use a procedure pointer: 

```Fortran
abstract interface
    subroutine blas_saxpy(blabla)
end interface

procedure(blas_saxpy), pointer :: axpy_ 
axpy_ => axpy
! Check that we're not using the internal implementation
call check(error, .not.associated(axpy_ , stdlib_saxpy))
```

But unfortunately, Fortran does not allow pointing to an interface, so this is not possible. If you have a better idea how could we check this, I'm very much in for removing preprocessing from the test module!

Reply:
Edit: the way that CMake uses to check if a fortran function exists is just to declare it `external` and try to compile a program:

https://github.com/Kitware/CMake/blob/a63bee680ce050e41f549ad9e3a811700347a3f6/Modules/CheckFortranFunctionExists.cmake#L62-L67

Reply:
I see, so if I understand, the objective of this test is to check if `STDLIB_EXTERNAL_BLAS` \ `STDLIB_EXTERNAL_LAPACK` are activated, that the hypothesis of an optimized library being called is fulfilled, right?

If say one does force these macros to be TRUE but the external library is not visible in the path, then a compile-time error would occur such as "'undefined reference to `dgetrf_'" which would already alert of a link issue. If the library is visible, then
```
use stdlib_linalg_lapack, only: getrf
...
call getrf( ... )
```
would have the same effect as the local test interface. So I'm just wondering about the pertinence of the current test as it is for stdlib as a provider of a higher-level interface? I do not have an answerer here

Now, if the test is kept as is, this file should be added here https://github.com/fortran-lang/stdlib/blob/4859081791fced582f3ca7b48a0c2235cc5c4845/config/fypp_deployment.py#L23 to enable the fpm job to treat it accordingly :)

Reply:
In the case of the stdlib CI, yes because there are testing routines that check that the base functions work anyways. 

But in general, if you don't build the tests there is no guarantee that these functions are available. 
Consider this example at the [Compiler Explorer](https://godbolt.org/z/cK5rM9Mrh): if you don't call `axpy`, the compiler doesn't care what's in the interface, so you're never sure. 

So long story short, I think we have these two options: 
- remove these two additional tests, but run a full stdlib CI with an external BLAS/LAPACK in at least one of the system configurations;
- keep these two tests, and then add preprocessing to the testing routines in the fpm deployment. 

So in light of the additional complexity, maybe let's just remove the tests and go for option 1?

Reply:
I would be in favor of option 1, so given this PR started by @zoziha is meant to help using the bindings in MSYS2, what about changing one of the windows CI jobs, I would also suggest one of the linux-intel jobs to link againts MKL. That should be more than good enough I think.